Fixes two issues:
 - Don't bundle file/libmagic
 - Avoid underlinking by linking against libz, libicuuc, and libicudata

--- extconf.rb.old	2012-05-27 11:17:30.798768657 +0200
+++ extconf.rb	2012-05-27 11:38:49.561899729 +0200
@@ -45,31 +45,12 @@
   exit(1)
 end
 
-##
-# libmagic dependency
-#
+# Fix underlinking
+have_library 'z' or abort 'libz missing'
+have_library 'icuuc' or abort 'libicuuc missing'
+have_library 'icudata' or abort 'libicudata missing'
 
-src = File.basename('file-5.08.tar.gz')
-dir = File.basename(src, '.tar.gz')
-
-Dir.chdir("#{CWD}/src") do
-  FileUtils.rm_rf(dir) if File.exists?(dir)
-
-  sys("tar zxvf #{src}")
-  Dir.chdir(dir) do
-    sys("./configure --prefix=#{CWD}/dst/ --disable-shared --enable-static --with-pic")
-    sys("make -C src install")
-    sys("make -C magic install")
-  end
-end
-
-FileUtils.cp "#{CWD}/dst/lib/libmagic.a", "#{CWD}/libmagic_ext.a"
-
-$INCFLAGS[0,0] = " -I#{CWD}/dst/include "
-$LDFLAGS << " -L#{CWD} "
-
-dir_config 'magic'
-unless have_library 'magic_ext' and have_header 'magic.h'
+unless have_library 'magic' and have_header 'magic.h'
   STDERR.puts "\n\n"
   STDERR.puts "***************************************************************************************"
   STDERR.puts "********* error compiling and linking libmagic. please report issue on github *********"
